generators/plugin/main.go

```diff
@@ -0,0 +1,584 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/dave/jennifer/jen"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	openAPIPath      = "openapi.json"
+	definitionsPath  = "./definitions"
+	openAPIPatchPath = definitionsPath + "/.openapi_patch.yml"
+)
+
+func main() {
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: ""})
+
+	err := exec()
+	if err != nil {
+		println(err.Error())
+		os.Exit(1)
+	}
+}
+
+func exec() error {
+	// Loads OpenAPIDoc with the patch
+	doc, err := readOpenAPIDoc(openAPIPath, openAPIPatchPath)
+	if err != nil {
+		return err
+	}
+
+	// Loads the definitions (yaml files)
+	definitions, err := listDefinitionFiles(definitionsPath)
+	switch {
+	case err != nil:
+		return fmt.Errorf("could not list definitions: %w", err)
+	case len(definitions) == 0:
+		return fmt.Errorf("no definitions found in %s", definitionsPath)
+	}
+
+	// Generates files
+	for _, fileName := range definitions {
+		log.Info().Str("file", fileName).Msg("generating")
+		err := genDefinition(doc, filepath.Join(definitionsPath, fileName))
+		if err != nil {
```

Comment:
nit: I guess here we can add the path to the failed configuration to the error, it is a tiny change but can make it handy during the navigation, when we'll have more configurations.
```go
		if err != nil {
			fullPath := filepath.Join(definitionsPath, fileName)
			return fmt.Errorf("%s:1:1: could not generate package: %w", fullPath, err)
		}
```

Reply:
done